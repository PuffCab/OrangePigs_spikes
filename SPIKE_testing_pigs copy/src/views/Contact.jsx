import React, { useState } from "react";

function Contact() {
  const [email, setEmail] = useState("");
  const [name, setName] = useState("");

  const [isSpanVisible, setIsSpanVisible] = useState(false);
  const showSpan = () => {
    // setIsSpanVisible(true);
    setTimeout(() => {
      setIsSpanVisible(true);
    }, 3000);
  };
  return (
    <>
      <h1>Contac</h1>
      <span
        data-testid="span"
        style={{ visibility: isSpanVisible ? "visible" : "hidden" }}
      >
        Hi {name}, message sent to {email}
      </span>
      <form action="">
        <input
          type="text"
          name="name"
          id="name"
          placeholder="name"
          value={name}
          onChange={(e) => setName(e.target.value)}
        />
        <input
          type="text"
          name="email"
          id="email"
          placeholder="email"
          value={email}
          onChange={(e) => setEmail(e.target.value)}
        />
      </form>
      <button data-testid="test-button" onClick={showSpan}>
        Send
      </button>
    </>
  );
}

export default Contact;
