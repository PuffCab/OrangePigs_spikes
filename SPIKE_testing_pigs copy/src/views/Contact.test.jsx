import { fireEvent, render, screen, waitFor } from "@testing-library/react";
import { expect, test } from "vitest";
import Contact from "./Contact";

test("header in Contact renders", () => {
  render(<Contact />);

  const h1 = screen.getByRole("heading");

  expect(h1).toBeInTheDocument();
});

test("should render two inputs with placerholder value email and name", () => {
  render(<Contact />);

  const nameInput = screen.getByPlaceholderText("name");
  const emailInput = screen.getByPlaceholderText("email");

  expect(nameInput, emailInput).toBeInTheDocument();
});
test("button element should render", () => {
  render(<Contact />);

  const button = screen.getByTestId("test-button");

  expect(button).toBeInTheDocument();
});

test("span element should not be initially visible", () => {
  render(<Contact />);

  const span = screen.getByTestId("span");

  expect(span).not.toBeVisible();
});

test("input elements should be initially empty", () => {
  render(<Contact />);

  const nameInput = screen.getByPlaceholderText("name");
  const emailInput = screen.getByPlaceholderText("email");

  expect(nameInput.value).toBe("");
  expect(emailInput.value).toBe("");
});

test("inputs should display typed value", () => {
  render(<Contact />);

  const nameInput = screen.getByPlaceholderText("name");
  const emailInput = screen.getByPlaceholderText("email");

  const nameInputText = "Tom";
  const emailInputText = "tom@tom.com";

  fireEvent.change(nameInput, { target: { value: nameInputText } });
  fireEvent.change(emailInput, { target: { value: emailInputText } });

  expect(nameInput.value).toBe(nameInputText);
  expect(emailInput.value).toBe(emailInputText);
});

test("span should display user name and email after clicking button", async () => {
  render(<Contact />);

  const nameInput = screen.getByPlaceholderText("name");
  const emailInput = screen.getByPlaceholderText("email");
  const button = screen.getByTestId("test-button");
  const span = screen.getByTestId("span");

  const nameInputText = "Tom";
  const emailInputText = "tom@tom.com";

  fireEvent.change(nameInput, { target: { value: nameInputText } });
  fireEvent.change(emailInput, { target: { value: emailInputText } });
  fireEvent.click(button);

  //   expect(span).toBeVisible();
  //   expect(span).toBeInTheDocument(
  //     `Hi ${nameInputText}, message sent to ${emailInputText}`
  //   );

  await waitFor(
    () => {
      expect(span).toBeVisible();
      expect(span).toBeInTheDocument(
        `Hi ${nameInputText}, message sent to ${emailInputText}`
      );
    },
    { timeout: 3000 }
  );
});

test("snapshot test", () => {
  const { container } = render(<Contact />);

  expect(container).toMatchSnapshot();
});
