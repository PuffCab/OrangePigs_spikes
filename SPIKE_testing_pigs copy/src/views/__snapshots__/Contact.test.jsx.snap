// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`snapshot test 1`] = `
<div>
  <h1>
    Contact
  </h1>
  <span
    data-testid="span"
    style="visibility: hidden;"
  >
    Hi 
    , message sent to 
  </span>
  <form
    action=""
  >
    <input
      id="name"
      name="name"
      placeholder="name"
      type="text"
      value=""
    />
    <input
      id="email"
      name="email"
      placeholder="email"
      type="text"
      value=""
    />
  </form>
  <button
    data-testid="test-button"
  >
    Send
  </button>
</div>
`;
